--- a/googletest/include/gtest/internal/gtest-port.h
+++ b/googletest/include/gtest/internal/gtest-port.h
@@ -1969,7 +1969,7 @@ inline bool IsXDigit(char32_t ch) {
 }
 inline bool IsXDigit(wchar_t ch) {
   const unsigned char low_byte = static_cast<unsigned char>(ch);
-  return ch == low_byte && isxdigit(low_byte) != 0;
+  return static_cast<unsigned char>(ch) == low_byte && isxdigit(low_byte) != 0;
 }
 
 inline char ToLower(char ch) {
